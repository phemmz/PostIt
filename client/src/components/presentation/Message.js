import React, { Component } from 'react';

class Message extends Component {
	render() {
		return (
			<div className="email-content-wrap">
				<div className="row">
					<div className="col s12 m10 l10 msg-position">
						<ul className="collection">
							<li className="collection-item avatar">
								<span className="circle green lighten-1 center">{this.props.currentMessage.messagecreator[0]}</span>
								<span className="email-title">{this.props.currentMessage.messagecreator}</span>
								<p className="grey-text ultra-small">Priority: {this.props.currentMessage.priority}</p>
								<p className="grey-text ultra-small">{new Date(this.props.currentMessage.createdAt).toLocaleString()}</p>
								<span className="email-content">{this.props.currentMessage.content}</span>
							</li>
						</ul>
					</div>
				</div>
				<hr />
			</div>
		)
	}
}
export default Message;