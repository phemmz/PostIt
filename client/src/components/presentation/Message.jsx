import React from 'react';
import PropTypes from 'prop-types';

const Message = ({
  currentMessage
}) => {
  return (
    <div className="email-content-wrap">
      <div className="row">
        <div className="col s12 m10 l10 msg-position">
          <ul className="collection">
            <li className="collection-item avatar">
              <span
                className="circle green lighten-1 center"
              >{currentMessage.messagecreator[0]}</span>
              <span
                className="email-title"
                id="msg-creator"
              >{currentMessage.messagecreator}</span>
              <span
                className="grey-text ultra-small right date-msg-position"
              >Priority: {currentMessage.priority}</span>
              <span
                className="grey-text ultra-small right date-pad date-msg-position"
              >{new Date(currentMessage.createdAt).toLocaleString()}</span>
              <div className="flex wordwrap">
                <span
                  className="email-content msg-content"
                >{currentMessage.content}</span>
              </div>
            </li>
          </ul>
        </div>
      </div>
      <hr />
    </div>
  );
};

Message.propTypes = {
  currentMessage: PropTypes.object.isRequired,
};

export default Message;
