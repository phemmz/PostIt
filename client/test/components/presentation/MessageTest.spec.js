import React from 'react';
import { mount } from 'enzyme';
import Message from '../../../src/components/presentation/Message';

describe('Message Component', () => {
  const currentMessage = {
    messagecreator: 'phemmz',
    priority: 'Normal',
    createdAt: '2017-09-06T14:53:09.414Z',
    content: 'nah'
  };

  const readList = 'phemmz';
  it('displays each message in a group', () => {
    const wrapper = mount(<Message
      currentMessage={currentMessage}
      readList={readList}
    />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.prop('currentMessage')).toBe(currentMessage);
    expect(wrapper.prop('readList')).toBe(readList);
    expect(wrapper.find('div').exists()).toBe(true);
    expect(wrapper.find('span').exists()).toBe(true);
    expect(wrapper.find('ul').exists()).toBe(true);
    expect(wrapper.find('li').exists()).toBe(true);
    expect(wrapper.find('i').exists()).toBe(true);
    expect(wrapper.find('hr').exists()).toBe(true);
    expect(wrapper.find('div').length).toBe(5);
    expect(wrapper.find('span').length).toBe(5);
    expect(wrapper.find('hr').length).toBe(1);
    expect(wrapper.find('ul').length).toBe(1);
    expect(wrapper.find('li').length).toBe(1);
    expect(wrapper.find('i').length).toBe(1);
  });
});
