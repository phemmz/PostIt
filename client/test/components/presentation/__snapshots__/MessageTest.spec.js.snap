// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Component displays each message in a group 1`] = `
<Message
  currentMessage={
    Object {
      "content": "nah",
      "createdAt": "2017-09-06T14:53:09.414Z",
      "messagecreator": "phemmz",
      "priority": "Normal",
    }
  }
  readList="phemmz"
>
  <div
    className="email-content-wrap"
  >
    <div
      className="row"
    >
      <div
        className="col s12 m10 l10 msg-position"
      >
        <ul
          className="collection my-collection"
        >
          <li
            className="collection-item avatar"
          >
            <span
              className="circle green lighten-1 center"
            >
              p
            </span>
            <span
              className="email-title"
              id="msg-creator"
            >
              phemmz
            </span>
            <span
              className="grey-text ultra-small right date-msg-position"
            >
              Priority: 
              Normal
            </span>
            <span
              className="grey-text ultra-small right date-pad date-msg-position"
            >
              9/6/2017, 3:53:09 PM
            </span>
            <div
              className="flex wordwrap"
            >
              <span
                className="email-content msg-content"
              >
                nah
              </span>
            </div>
            <div
              className="right"
            >
              <i
                className="material-icons read-marker-done tooltipped"
                data-tooltip="Read by: phemmz"
              >
                done_all
              </i>
            </div>
          </li>
        </ul>
      </div>
    </div>
    <hr />
  </div>
</Message>
`;
